### DISCUSSION TITLE
iOS 开发中使用 NSURLProtocol 拦截 HTTP 请求 · /intercept

### DISCUSSION THREAD

**Discussion by draveness** [draveness/blog-comments#277] [Announcements]
https://draveness.me/intercept

**Comment by KelyCocoa** [comment]
大神 我这两个地方时常出现也指针崩溃 请大神指点， 跪谢

\# 11870 SIGSEGV
SEGV_ACCERR
chazuoxy -[CZCustomURLProtocol URLSession:dataTask:didReceiveData:] (CZCustomURLProtocol.m:)

- (void)URLSession:(NSURLSession *)session dataTask:(NSURLSessionDataTask *)dataTask didReceiveData:(NSData *)data {
	
	[self.client URLProtocol:self didLoadData:data];
}

- (void)URLSession:(NSURLSession *)session task:(NSURLSessionTask *)task didCompleteWithError:(NSError *)error {
	
	if (error) {
		[self.client URLProtocol:self didFailWithError:error];
	} else {
		[self.client URLProtocolDidFinishLoading:self];
	}
}
